### PR TITLE
[USM] Add envoy configuration value

### PR DESCRIPTION
**Pull request by amitslavin**
### What does this PR do?

Add configuration value for `envoyCmd`.

### Motivation

Fix the issue where the Envoy path is not set to the default value, causing Istio support to fail.

### CODE REVIEW

File: pkg/config/setup/system_probe.go [hunk 1]
```diff
@@ -234,6 +234,7 @@ func InitSystemProbeConfig(cfg pkgconfigmodel.Config) {
 	cfg.BindEnvAndSetDefault(join(smNS, "enable_kafka_monitoring"), false)
 	cfg.BindEnv(join(smNS, "enable_postgres_monitoring"))
 	cfg.BindEnvAndSetDefault(join(smNS, "tls", "istio", "enabled"), false)
+	cfg.BindEnvAndSetDefault(join(smNS, "tls", "istio", "envoy_path"), "")
```
Comment (guyarb) [inline pkg/config/setup/system_probe.go:237]:
```suggestion
	cfg.BindEnv(join(smNS, "tls", "istio", "envoy_path"), "")
```

File: pkg/network/usm/istio.go [hunk 2]
```diff
@@ -285,6 +287,10 @@ func (m *istioMonitor) getEnvoyPath(pid uint32) string {
 	}
 
 	buffer = buffer[:n]
+
+	if m.envoyPath != "" {
+		envoyCmd = []byte(m.envoyPath)
+	}
```
Comment (guyarb) [inline pkg/network/usm/istio.go:293]: you should do that in the constructor, and do it only once
if the config is empty, then set the default option into `m.envoyPath`

**Review by guyarb — COMMENTED** [review]
Overall, that's a good change
I'm missing some documentation and reasoning

What about better path resolution? Do you defer it for a followup PR?

File: pkg/network/usm/istio.go [hunk 3]
```diff
@@ -115,11 +115,14 @@ func newIstioMonitor(c *config.Config, mgr *manager.Manager) *istioMonitor {
 	}
 
 	procRoot := kernel.ProcFSRoot()
+	if c.EnvoyPath != "" {
+		envoyCmd = []byte(c.EnvoyPath)
```
Comment (guyarb) [inline pkg/network/usm/istio.go:119]: Don't override the global
although no bad impact will happen (as we have a single instance of istio monitoring)
but, that's not future proof
I can image having other "envoy monitors", and overriding envoyCmd can cause bad outcomes

**Comment by amitslavin** [comment]
Issue is being handled in the following PR: https://github.com/DataDog/datadog-agent/pull/27400

**Pull request closed without merging**